File: mobile/shared/src/commonMain/kotlin/com/bissbilanz/model/Stats.kt
package com.bissbilanz.model

import kotlinx.serialization.Serializable

@Serializable
data class MacroTotals(
    val calories: Double = 0.0,
    val protein: Double = 0.0,
    val carbs: Double = 0.0,
    val fat: Double = 0.0,
    val fiber: Double = 0.0,
)

@Serializable
data class DailyStatsResponse(
    val data: List<DailyStatsEntry>,
    val goals: Goals? = null,
)

@Serializable
data class WeeklyMonthlyStatsResponse(
    val stats: MacroTotals,
)

@Serializable
data class MealBreakdownResponse(
    val data: List<MealBreakdownEntry>,
)

@Serializable
data class TopFoodsResponse(
    val data: List<TopFoodEntry>,
)

@Serializable
data class CalendarDay(
    val date: String,
    val calories: Double,
    val goalMet: Boolean = false,
    val hasEntries: Boolean,
)

@Serializable
data class CalendarDayRaw(
    val calories: Double = 0.0,
    val hasEntries: Boolean = false,
)

@Serializable
data class CalendarResponse(
    val days: Map<String, CalendarDayRaw>,
)

@Serializable
data class CopyEntriesResponse(
    val entries: List<Entry>,
    val count: Int,
)

@Serializable
data class MaintenanceResult(
    val maintenanceCalories: Double,
    val dailyDeficitOrSurplus: Double,
    val weightChangeKg: Double,
    val fatMassChangeKg: Double,
    val muscleMassChangeKg: Double,
    val fatEnergyKcal: Double,
    val muscleEnergyKcal: Double,
    val avgDailyCalories: Double,
)

@Serializable
data class MaintenanceResponse(
    val result: MaintenanceResult,
    val meta: MaintenanceMeta,
)

@Serializable
data class MealTypeCreate(
    val name: String,
)

@Serializable
data class MealTypesResponse(
    val mealTypes: List<MealType>,
)

File: mobile/iosApp/PhoneShared/Models/Stats.swift
import Foundation

struct MacroTotals: Codable {
    let calories: Double
    let protein: Double
    let carbs: Double
    let fat: Double
    let fiber: Double

    static let zero = MacroTotals(calories: 0, protein: 0, carbs: 0, fat: 0, fiber: 0)
}

struct DailyStatsEntry: Codable {
    let date: String
    let calories: Double
    let protein: Double
    let carbs: Double
    let fat: Double
    let fiber: Double
}

struct DailyStatsResponse: Codable {
    let data: [DailyStatsEntry]
    let goals: Goals?
}

struct WeeklyMonthlyStatsResponse: Codable {
    let stats: MacroTotals
}

struct MealBreakdownEntry: Codable {
    let mealType: String
    let calories: Double
    let protein: Double
    let carbs: Double
    let fat: Double
    let fiber: Double
}

struct MealBreakdownResponse: Codable {
    let data: [MealBreakdownEntry]
}

struct StreaksResponse: Codable {
    let currentStreak: Int
    let longestStreak: Int
}

struct TopFoodEntry: Codable, Identifiable {
    let foodId: String?
    let recipeId: String?
    let foodName: String
    let count: Int
    let calories: Double
    let protein: Double
    let carbs: Double
    let fat: Double
    let fiber: Double

    var id: String {
        foodId ?? recipeId ?? foodName
    }
}

struct TopFoodsResponse: Codable {
    let data: [TopFoodEntry]
}

struct CalendarDay: Codable {
    let date: String
    let calories: Double
    let hasGoal: Bool
    let metGoal: Bool
}

/// Wire format of GET /api/stats/calendar: a map keyed by yyyy-MM-dd holding
/// only calories and hasEntries — goal state is computed client-side.
struct CalendarDayData: Codable {
    let calories: Double
    let hasEntries: Bool
}

struct CalendarResponse: Codable {
    let days: [String: CalendarDayData]
}

extension CalendarDay {
    static func days(from wire: [String: CalendarDayData], calorieGoal: Double?) -> [CalendarDay] {
        wire.map { date, day in
            CalendarDay(
                date: date,
                calories: day.calories,
                hasGoal: calorieGoal != nil,
                metGoal: calorieGoal.map { day.calories > 0 && day.calories <= $0 } ?? false
            )
        }
        .sorted { $0.date < $1.date }
    }
}

struct MaintenanceRequest: Codable {
    let startDate: String
    let endDate: String
    let bodyFatChangeRatio: Double?
}

struct MaintenanceResponse: Codable {
    let maintenanceCalories: Double
    let avgDailyCalories: Double
    let dailyDeficitSurplus: Double
    let weightChange: Double
    let startWeight: Double
    let endWeight: Double
    let totalDays: Int
    let weightEntryCount: Int
    let foodEntryDays: Int
    let coveragePercent: Double
    let fatChange: Double?
    let muscleChange: Double?
}

struct SupplementHistoryResponse: Codable {
    let history: [SupplementHistoryItem]
}

struct SupplementHistoryItem: Codable, Identifiable {
    let supplementId: String
    let supplementName: String
    let date: String
    let takenAt: String

    /// Synthetic id so SwiftUI can diff: one entry per (supplement, day).
    var id: String {
        "\(supplementId)-\(date)"
    }
}

/// Alias retained for existing call sites that read `.history` as a list of "entries".
typealias SupplementHistoryEntry = SupplementHistoryItem

struct MealTypeCreate: Codable {
    let name: String
}

struct MealTypesResponse: Codable {
    let mealTypes: [MealType]
}

struct MealTypeResponse: Codable {
    let mealType: MealType
}
